client/landing/stepper/declarative-flow/internals/steps-repository/goals/index.tsx

```diff
@@ -223,6 +248,11 @@ const GoalsStep: StepType = ( { navigation, flow } ) => {
 				nextLabelText={ translate( 'Next' ) }
 				skipLabelText={ translate( 'Skip' ) }
 				recordTracksEvent={ recordTracksEvent }
+				goBack={ () => {
+					const destination = addQueryArgs( withLocale( '/setup/onboarding/plans', locale ) );
+					window.location.replace( destination );
+				} }
+				hideBack={ ! site?.plan?.is_free }
```

Comment:
Should this be decided by the framework instead? Meaning, a back button would be displayed only if there is a previous step in the current flow?

Reply:
The issue here is that pressing back goes to a different flow, the goals are part of the site setup flow and are the first step, there is no previous step, we want pressing back to go to a different flow for the onboarding in the plans section.